public/components/event_analytics/explorer/events_views/data_grid.tsx

```diff
@@ -65,14 +70,18 @@ export function DataGrid(props: DataGridProps) {
     pplService,
     requestParams,
   });
+
+  const query = useSelector(selectQueries)[tabId];
```

Comment:
Seems like all you need related to tabId is for getting redux query state, you can use context. the tabId has been already exposed. tabId is not required as props for data grid I believe.

Reply:
changed